### ISSUE TITLE
进入 train_step() 方法查看梯度更新时，parms.grad 值为None

### ISSUE THREAD

**Issue by qqqtwh** [open-mmlab/mmdetection#11234]
在 mmengine>model>base_model>base_model.py 的 BaseModel类中，进入 train_step() 方法查看梯度更新时，parms.grad 值为None

  def train_step(self, data: Union[dict, tuple, list],optim_wrapper: OptimWrapper) -> Dict[str, torch.Tensor]:

        with optim_wrapper.optim_context(self):
                data = self.data_preprocessor(data, True)
                losses = self._run_forward(data, mode='loss')  # type: ignore
        parsed_losses, log_vars = self.parse_losses(losses)  # type: ignore

        for name,parms in self.named_parameters():
            print('-->name:', name)
            print('-->para:', parms[0,0,0,:])
            print('-->grad_requirs:',parms.requires_grad)
            print('-->grad_value:',parms.grad)
            print("===",__file__)
            break
        
        optim_wrapper.update_params(parsed_losses)
        
        for name,parms in self.named_parameters():
            print('-->name:', name)
            print('-->para:', parms[0,0,0,:])
            print('-->grad_requirs:',parms.requires_grad)
            print('-->grad_value:',parms.grad)                        # 此处结果为None
            print("===",__file__)
            break
        
        return log_vars

**Comment by Tiwn** [comment]
我也遇到了这个问题，请问你解决了吗？可以传授传授经验吗，太感谢了

**Comment by Tiwn** [comment]
找到原因了，优化器每次step后就紧接着进行zero_gard了
